python/ray/train/torch.py

```diff
@@ -504,7 +504,8 @@ def _wait_for_batch(self, item):
         # the tensor might be freed once it is no longer used by
         # the creator stream.
         for i in item:
-            i.record_stream(curr_stream)
+            if isinstance(i, torch.Tensor):
```

Comment:
can you comment what may show up here as well, and why you need this if statement now?

Reply:
The pytorch dataloader can actually just return a batch of anything. In all of our examples and tests our data loaders return batches of tensors, but in this case, the torch geometric data loader also returns batch size, node id, etc., which are not all tensors.